packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -741,4 +745,82 @@ export class AbstractQueryInterfaceTypeScript<Dialect extends AbstractDialect =
 
     return this.sequelize.queryRaw(sql, { ...bulkDeleteOptions, raw: true, type: QueryTypes.DELETE });
   }
+
+  /**
+   * Update multiple records of a table
+   *
+   * @example
+   * queryInterface.bulkUpdate('roles', {
+   *     label: 'admin',
+   *   }, {
+   *     userType: 3,
+   *   },
+   * );
+   *
+   * @param tableName         Table name to update
+   * @param values            Values to be inserted, mapped to field name
+   * @param options           Various options, please see Model.bulkCreate options
+   * @param columnDefinitions Attributes on return objects if supported by SQL dialect
+   */
+  async bulkUpdate(
+    tableName: TableNameOrModel,
+    values: Record<string, unknown>,
+    options?: QiUpdateOptions,
+    columnDefinitions?: Record<string, AttributeOptions>,
+  ): Promise<number> {
```

Comment:
bulkUpdate supports `RETURNING` so it should not return a number but `[updatedCount, updatedValues]`

`updatedValues` should be a plain object with the values returned from the database, not a model instance

Reply:
See comment below